contrib/labstack/echo/echotrace.go

```diff
@@ -0,0 +1,49 @@
+// Package echo provides functions to trace the labstack/echo package (https://github.com/labstack/echo).
+package echo
+
+import (
+	"strconv"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+
+	"github.com/labstack/echo"
+)
+
+// Middleware returns middleware that will trace incoming requests.
+func Middleware(service string) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			request := c.Request()
+			resource := c.Path()
+			opts := []ddtrace.StartSpanOption{
+				tracer.ServiceName(service),
+				tracer.ResourceName(resource),
+				tracer.SpanType(ext.SpanTypeWeb),
+				tracer.Tag(ext.HTTPMethod, request.Method),
+				tracer.Tag(ext.HTTPURL, request.URL.Path),
+			}
+
+			if spanctx, err := tracer.Extract(tracer.HTTPHeadersCarrier(request.Header)); err == nil {
+				opts = append(opts, tracer.ChildOf(spanctx))
+			}
+			span, ctx := tracer.StartSpanFromContext(request.Context(), "http.request", opts...)
+			defer span.Finish()
+
+			// pass the span through the request context
+			c.SetRequest(request.WithContext(ctx))
+
+			// serve the request to the next middleware
+			err := next(c)
+
+			span.SetTag(ext.HTTPCode, strconv.Itoa(c.Response().Status))
+			if err != nil {
+				span.SetTag("echo.errors", err.Error())
```

Comment:
What's the point to have this tag? We already have the standard error.

Reply:
you are right. i'll remove it.